provisioner/provision.js

```diff
@@ -192,9 +194,50 @@ Provisioner.prototype.runAllProvisionersOnce = async function () {
   debug('configured workers:           %j', workerNames);
   debug('managed requests/instances:   %j', this.awsManager.knownWorkerTypes());
 
-  return Promise.all(workerTypes.map(async worker => {
-    return this.provisionType(worker, pricing);
-  }));
+  var forSpawning = [];
+
+  for (var worker of workerTypes) {
+    var change = await this.changeForType(worker, pricing);
+
+    if (change > 0) {
+      debug('%s needs %d capacity created', worker.workerType, change);
+      var bids = worker.determineSpotBids(this.awsManager.ec2.regions, pricing, change);
+      // This could probably be done cleaner
+      for (let bid of bids) {
+        forSpawning.push({workerType: worker, bid: bid});
+      }
+    } else if (change < 0) {
+      var capToKill = -change;
+      debug('%s needs %d capacity destroyed', worker.workerType, capToKill);
+      await this.awsManager.killCapacityOfWorkerType(
+            worker, capToKill, ['pending', 'spotReq']);
+    } else {
+      debug('%s needs no changes', worker.workerType);
+    }
+  }
+
+  var d = delayer(500);
+  var longD = delayer(2000);
+
+  // We want to have a maximum number of attempts
+  var attemptsLeft = forSpawning.length * 2;
```

Comment:
so what i'm doing is getting a list of things to spawn, ordered by workerType.  I'm then shuffling them up and so I no longer have them separated by workerType.  I would have to maintain a retry count per workerTYpe, which isn't important.  Instead, I decided to just say that you have 2x the total number of all things to spawn attempts before failing.

Not perfect, but we're going to retry in 75s anyway...